### ISSUE TITLE
can buy pinned off-season avatar transformation buffs (Snowball, Shiny Seed, Seafoam, Spooky Sparkles)

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9899]
If you pin an avatar transformation buff to your Rewards list, then you can buy it at any time of the year, even when it's no longer available in the Shop.

The desired behaviour is both of these:
- The API should prohibit buying the buff outside its allowed time period
- If it's been pinned, it should become unpinned or invisible outside its allowed time period

**Comment by kartikadur** [comment]
Since I've been working on the task page for other stuff can I take this as well?

**Comment by Alys** [comment]
@kartikadur That would be awesome!

**Comment by kartikadur** [comment]
Is there anything I should know before starting on this?

**Comment by Alys** [comment]
@kartikadur I can't think of anything, but comment if you can't find the code you need.

**Comment by kartikadur** [comment]
What part of the code is responsible for adding / updating transformation buffs? Is there a cron script or are these manually added to the database?

**Comment by kartikadur** [comment]
At the end of the season should the special gear be only hidden or completely removed?

**Comment by Alys** [comment]
My feeling is that hidden would be best. The advantage of leaving it pinned but hidden is that we don't have to run a migration to unpin it, and it's still pinned for the user next year when it's available again. If they've pinned it once, they're likely to be still interested the next year.

I'm not sure about your previous question about how the transformation buffs are enabled each season. @SabreCat ?

**Comment by SabreCat** [comment]
It's currently a manual process. The available transformation items are set up in https://github.com/HabitRPG/habitica/blob/develop/website/common/script/libs/shops-seasonal.config.js under `availableSpells`.

**Comment by beffymaroo** [comment]
@kartikadur Are you still working on this issue? Please let us know. Thank you!

**Comment by kartikadur** [comment]
@beffymaroo I have a few pressing concerns to deal with at the moment. I hope to get around to it by the end of this month.

**Comment by beffymaroo** [comment]
@kartikadur Just checking in on this issue as well. Let us know how it's going when you have time. Thank you!

**Comment by kartikadur** [comment]
I'm dealing with the logistics of shifting to a new job, so I wont be able
to deal with this for a few months at best. If someone else wants to work
on this let them have it. I'll work on something else when I have time in
the near future.

**Comment by SabreCat** [comment]
(We've been addressing this by removing pinned items in the database when the item becomes unavailable in the shop, so it's not an urgent issue.)

**Comment by kartikadur** [comment]
*Thanks for understanding.*

**Comment by shanaqui** [comment]
> (We've been addressing this by removing pinned items in the database when the item becomes unavailable in the shop, so it's not an urgent issue.)

This seems not to have worked this time, maybe? From RaB:

> Go2space   @Go2space (7452eebe-7d07-4cc7-8daa-65a6233dc4d2) [web] 
> So idk if this is a bug, but I pinned snowballs and new year's cards during the winter solstice and they're still in my rewards column. Is this normal? I checked and I am indeed still able to purchase them.

**Comment by SabreCat** [comment]
Possibly missed. I've created a seasonal event checklist, will make sure that pinned item removal is on there for end-of-event processing!

**Comment by Alys** [comment]
I've removed some of the older posts from here that were just about the issue being claimed so that the ones containing technical details are easier to find.
This issue is still occurring and is open for someone to work on.

**Comment by veeeeeee** [comment]
On hold for now--we'd like to make this be consistent with other seasonal/limited timing items that can be pinned.

**Comment by SabreCat** [comment]
Fix coming out with 46b8e7f7 later this week. Closing!
